### ISSUE TITLE
`ipfs get`: tar has invalid root when a filename contains double dot `..`

### ISSUE THREAD

**Issue by karmanyaahm** [ipfs/go-ipfs#8293]
### Checklist

- [X] This is a bug report, not a question. Ask questions on [discuss.ipfs.io](https://discuss.ipfs.io).
- [X] I have searched on the [issue tracker](https://github.com/ipfs/go-ipfs/issues?q=is%3Aissue) for my bug.
- [X] I am running the latest [go-ipfs version](https://dist.ipfs.io/#go-ipfs) or have an issue updating.

### Installation method

ipfs-update or dist.ipfs.io

### Version

```Text
go-ipfs version: 0.9.1
Repo version: 11
System version: amd64/linux
Golang version: go1.16.6
```

### Config

```json
{
  "API": {
    "HTTPHeaders": {}
  },
  "Addresses": {
    "API": "/ip4/127.0.0.1/tcp/5001",
    "Announce": [],
    "Gateway": "/ip4/127.0.0.1/tcp/4002",
    "NoAnnounce": [],
    "Swarm": [
      "/ip4/0.0.0.0/tcp/4001",
      "/ip6/::/tcp/4001",
      "/ip4/0.0.0.0/udp/4001/quic",
      "/ip6/::/udp/4001/quic"
    ]
  },
  "AutoNAT": {},
  "Bootstrap": [
    "/dns6/karmanyaah.malhotra.cc/tcp/4001/p2p/12D3KooWAaaAAPjJMAdXtv6i4FR6wn1N51f1wA1iej3XeJkwcRth",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb",
    "/dnsaddr/bootstrap.libp2p.io/p2p/QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt",
    "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
    "/ip4/104.131.131.82/udp/4001/quic/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
    "/dns6/unifiedpush.org/tcp/4001/p2p/12D3KooWQP2mXMcTRPUqgNdubydN6ucAghYHDCRni8XZosNjTx3q"
  ],
  "Datastore": {
    "BloomFilterSize": 0,
    "GCPeriod": "1h",
    "HashOnRead": false,
    "Spec": {
      "mounts": [
        {
          "child": {
            "path": "blocks",
            "shardFunc": "/repo/flatfs/shard/v1/next-to-last/2",
            "sync": true,
            "type": "flatfs"
          },
          "mountpoint": "/blocks",
          "prefix": "flatfs.datastore",
          "type": "measure"
        },
        {
          "child": {
            "compression": "none",
            "path": "datastore",
            "type": "levelds"
          },
          "mountpoint": "/",
          "prefix": "leveldb.datastore",
          "type": "measure"
        }
      ],
      "type": "mount"
    },
    "StorageGCWatermark": 90,
    "StorageMax": "10GB"
  },
  "Discovery": {
    "MDNS": {
      "Enabled": true,
      "Interval": 10
    }
  },
  "Experimental": {
    "FilestoreEnabled": false,
    "GraphsyncEnabled": false,
    "Libp2pStreamMounting": false,
    "P2pHttpProxy": false,
    "ShardingEnabled": false,
    "StrategicProviding": false,
    "UrlstoreEnabled": false
  },
  "Gateway": {
    "APICommands": [],
    "HTTPHeaders": {
      "Access-Control-Allow-Headers": [
        "X-Requested-With",
        "Range",
        "User-Agent"
      ],
      "Access-Control-Allow-Methods": [
        "GET"
      ],
      "Access-Control-Allow-Origin": [
        "*"
      ]
    },
    "NoDNSLink": false,
    "NoFetch": false,
    "PathPrefixes": [],
    "PublicGateways": {
      "karmanyaah.malhotra.cc": {
        "NoDNSLink": false,
        "Paths": [
          "/ipns"
        ],
        "UseSubdomains": false
      }
    },
    "RootRedirect": "",
    "Writable": false
  },
  "Identity": {
    "PeerID": "12D3KooWGZtnCxVUqC4qhjRMhzvN2pEm1sh3Vy1Z8L5uKVnVW6ap"
  },
  "Ipns": {
    "RecordLifetime": "2400h",
    "RepublishPeriod": "",
    "ResolveCacheSize": 128
  },
  "Mounts": {
    "FuseAllowOther": false,
    "IPFS": "/ipfs",
    "IPNS": "/ipns"
  },
  "Peering": {
    "Peers": null
  },
  "Pinning": {},
  "Plugins": {
    "Plugins": null
  },
  "Provider": {
    "Strategy": ""
  },
  "Pubsub": {
    "DisableSigning": false,
    "Router": ""
  },
  "Reprovider": {
    "Interval": "12h",
    "Strategy": "all"
  },
  "Routing": {
    "Type": "dht"
  },
  "Swarm": {
    "AddrFilters": null,
    "ConnMgr": {
      "GracePeriod": "20s",
      "HighWater": 900,
      "LowWater": 600,
      "Type": "basic"
    },
    "DisableBandwidthMetrics": false,
    "DisableNatPortMap": false,
    "EnableAutoRelay": false,
    "EnableRelayHop": false,
    "Transports": {
      "Multiplexers": {},
      "Network": {},
      "Security": {}
    }
  }
}
```

### Description

```
$ echo abc > 'S.M.A.R.T..html'
$ ipfs add --pin=false 'S.M.A.R.T..html' --wrap-with-directory
added Qme4u9HfFqYUhH4i34ZFBKi1ZsW7z4MYHtLxScQGndhgKE S.M.A.R.T..html
added QmZnd7vpUVX9Shr1nYazH6CFFq8DpzEwaHgvoXvADMtG9a 
 4 B / 4 B [========] 100.00%

$ ipfs get QmZnd7vpUVX9Shr1nYazH6CFFq8DpzEwaHgvoXvADMtG9a
Saving file(s) to QmZnd7vpUVX9Shr1nYazH6CFFq8DpzEwaHgvoXvADMtG9a
 73 B / 73 B [========] 100.00% 0s
Error: contains more than one root or the root directory is not the first entry : tar has invalid root
```
Using v0.8.0 works fine.
I'm running into this bug because the Arch Wiki has a page called `S.M.A.R.T..html` and I run an IPFS mirror at `/ipns/archwiki.karmanyaah.malhotra.cc`.

**Comment by karmanyaahm** [comment]
This seems to happen post c47fca7d

**Comment by aschmahmann** [comment]
Thanks for the report, we'll take a look. In the meanwhile you can use `ipfs get -a myarchive.tar` and then just untar the result.

**Comment by petar** [comment]
Resolved by https://github.com/ipfs/tar-utils/pull/5

**Comment by LynHyper** [comment]
@aschmahmann Has the dependency been merged in v0.11/v0.12?

**Comment by aschmahmann** [comment]
Yep, it was merged in #8571. Post back if you have any issues.
